components/dashboard/UpcomingInterviewDialog/UpcomingInterviewDialog.js

```diff
@@ -0,0 +1,250 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import firebase from 'firebase/app';
+import DateFnsUtils from '@date-io/date-fns';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import Typography from '@material-ui/core/Typography';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+import FormControl from '@material-ui/core/FormControl';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import MenuItem from '@material-ui/core/MenuItem';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Box from '@material-ui/core/Box';
+import { makeStyles } from '@material-ui/core/styles';
+import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
+
+import upcomingInterviewFormValidation from './UpcomingInterviewValidationRules';
+import useFormValidation from './formValidationHook';
+import { useAuth } from '../../Auth';
+import SubmitSuccess from '../SubmitSuccess';
+import { DialogContent, DialogTitle, DialogActions } from '../../DialogComponents';
+
+const INTERVIEW_TYPES = [
+  {
+    value: 'phone-screen',
+    label: 'Phone Interview',
+    description:
+      'A phone interview, also known as a phone screen, is a short interview conducted over the telephone, which is often used to screen candidates before in-person interviews.',
+  },
+  {
+    value: 'recorded-video',
+    label: 'Recorded Video Interview',
+    description:
+      'A recorded video interview, also known as a one-way interview, is an interview where the interviewer is not present when the candidate records their answers to preset questions.',
+  },
+  {
+    value: 'live-video',
+    label: 'Live Video Interview',
+    description:
+      'A live video interview connects the interviewer and candidate face-to-face on video via webcam.',
+  },
+  {
+    value: 'in-person',
+    label: 'In-person Interview',
+    description:
+      'An in-person interview is a face-to-face meeting between the candidate and the interviewer.',
+  },
+];
+
+const useStyles = makeStyles(theme => ({
+  placeholder: {
+    color: 'currentColor',
+    opacity: theme.palette.type === 'light' ? 0.42 : 0.5,
+    transition: theme.transitions.create('opacity', {
+      duration: theme.transitions.duration.shorter,
+    }),
+  },
+  selectItem: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    flexDirection: 'row',
+    maxWidth: theme.breakpoints.width('sm'),
```

Comment:
Clever!

What do you think of this, going a step further: https://github.com/newjersey/career-network/pull/89/commits/db97e9629e4c60353a6ebf9ad4ab87f640d6256c